pkg/backend/metrics.go

```diff
@@ -0,0 +1,203 @@
+}
+
+func (ocb *openShiftClusterBackend) getResultType(backendErr error) utillog.ResultType {
+	var resultType utillog.ResultType
+	err, ok := backendErr.(*api.CloudError)
+	if ok {
+		resultType = utillog.MapStatusCodeToResultType(err.StatusCode)
+	}
+	return resultType
+}
+
+func (ocb *openShiftClusterBackend) logMetricDimensions(log *logrus.Entry, dimensions map[string]string) {
+	for metric, value := range dimensions {
+		log.Info(fmt.Sprintf("%s: %s = %s", metricPackage, metric, value))
+	}
+}
+
+func (m *openShiftClusterBackend) gatherCorrelationID(doc *api.OpenShiftClusterDocument, dimensions map[string]string) {
+	if doc.CorrelationData != nil {
+		dimensions["correlationid"] = doc.CorrelationData.CorrelationID
+		dimensions["client_requestid"] = doc.CorrelationData.ClientRequestID
+		dimensions["requestid"] = doc.CorrelationData.RequestID
+		dimensions["async_operationsid"] = doc.AsyncOperationID
+	} else {
+		dimensions["correlationid"] = empty
+		dimensions["client_requestid"] = empty
+		dimensions["async_operationsid"] = empty
+	}
+}
+
+func (ocb *openShiftClusterBackend) gatherMiscMetrics(doc *api.OpenShiftClusterDocument, dimensions map[string]string) {
+	dimensions["subscriptionid"] = ocb.env.SubscriptionID()
+	dimensions["resourceid"] = doc.ResourceID
+	dimensions["location"] = doc.OpenShiftCluster.Location
+	dimensions["resourcegroup"] = doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID
+	dimensions["name"] = doc.OpenShiftCluster.Name
+	dimensions["ocpversion"] = doc.OpenShiftCluster.Properties.ClusterProfile.Version
+	dimensions["workercount"] = strconv.FormatInt(int64(len(doc.OpenShiftCluster.Properties.WorkerProfiles)), 10)
+	dimensions["rpversion"] = doc.OpenShiftCluster.Properties.ProvisionedBy
+
+	for flag, feature := range doc.OpenShiftCluster.Properties.OperatorFlags {
+		dimensions[fmt.Sprintf("operator-%s", flag)] = feature
+	}
+
+	if doc.OpenShiftCluster.Tags != nil {
+		dimensions["tags"] = enabled
+	} else {
+		dimensions["tags"] = disabled
+	}
+}
+
+func (ocb *openShiftClusterBackend) gatherNodeMetrics(doc *api.OpenShiftClusterDocument, dimensions map[string]string) {
+	if doc.OpenShiftCluster.Properties.MasterProfile.DiskEncryptionSetID != "" {
+		dimensions["masterprofile.diskencryptionsetid"] = enabled
+	} else {
+		dimensions["masterprofile.diskencryptionsetid"] = disabled
+	}
+
+	if doc.OpenShiftCluster.Properties.MasterProfile.EncryptionAtHost == api.EncryptionAtHostEnabled {
+		dimensions["masterprofile.encryptionathost"] = string(api.EncryptionAtHostEnabled)
+	} else if doc.OpenShiftCluster.Properties.MasterProfile.EncryptionAtHost == api.EncryptionAtHostDisabled {
+		dimensions["masterprofile.encryptionathost"] = string(api.EncryptionAtHostDisabled)
+	} else {
+		dimensions["masterprofile.encryptionathost"] = unknown
+	}
+
+	for _, p := range doc.OpenShiftCluster.Properties.WorkerProfiles {
+		if p.VMSize != api.VMSizeStandardD4sV3 {
+			dimensions["customvmsize"] = string(p.VMSize)
```

Comment:
IMO we should just emit the VM size to a generic `workervmsize` dimension, and not worry about whether it's the "default" or not here. 

We should also emit the VM size chosen for the control plane as a dimension here too.

Reply:
I update this to simply set the vmsize for workers and masters.